bindings/python/unicorn/arch/arm.py

```diff
@@ -0,0 +1,59 @@
+# AArch32 classes and structures.
+#
+# @author elicn
+
+from typing import Any, Tuple
+
+import ctypes
+
+from .. import Uc, UcError
+from .. import arm_const as const
+from ..unicorn_const import UC_ERR_ARG
+
+ARMCPReg = Tuple[int, int, int, int, int, int, int]
+ARMCPRegValue = Tuple[int, int, int, int, int, int, int, int]
+
+class UcRegCP(ctypes.Structure):
+    """ARM coprocessors registers for instructions MRC, MCR, MRRC, MCRR
+    """
+
+    _fields_ = (
+        ('cp',   ctypes.c_uint32),
+        ('is64', ctypes.c_uint32),
+        ('sec',  ctypes.c_uint32),
+        ('crn',  ctypes.c_uint32),
+        ('crm',  ctypes.c_uint32),
+        ('opc1', ctypes.c_uint32),
+        ('opc2', ctypes.c_uint32),
+        ('val',  ctypes.c_uint64)
+    )
+
+    @property
+    def value(self) -> int:
+        return self.val
+
+    @classmethod
+    def from_param(cls, param: ARMCPRegValue):
+        assert type(param) is tuple and len(param) == len(cls._fields_)
+
+        return cls(*param)
+
+
+class UcAArch32(Uc):
```

Comment:
As stated further in the review below, my thought is that, such the arch-specific class might be used as a member of the `Uc` object. This in fact is `pimpl` design instead of your `factory` design if I understand it correctly.